File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/iterator/pagination/Page.java
package com.bigboxcode.designpattern.iterator.pagination;

public class Page {
    private int number;
    private String path;


    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public String getPath() {
        if (path == null) {
            return "/page/" + number;
        }

        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: design-patterns/php/iterator/pagination/Page.php
<?php
// Page.php

namespace BigBoxCode\DesignPattern\Iterator\Pagination;

class Page {
    private int $pageNumber = 0;
    private ?string $path = null;

    public function getNumber(): int {
        return $this->pageNumber;
    }

    public function setNumber(int $pageNumber): void {
        $this->pageNumber = $pageNumber;
    }

    public function getPath(): string {
        if ($this->path == null) {
            return "/page/" . $this->pageNumber;
        }

        return $this->path;
    }

    public function setPath(string $path): void {
        $this->path = $path;
    }
}